src/pages/settings/Troubleshoot/TroubleshootPage.tsx

```diff
@@ -65,15 +65,31 @@ function TroubleshootPage() {
             ExportOnyxState.shareAsFile(JSON.stringify(dataToShare));
         });
     }, [shouldMaskOnyxState]);
-
     const surveyCompletedWithinLastMonth = useMemo(() => {
         const surveyThresholdInDays = 30;
-        if (!tryNewDot?.classicRedirect?.timestamp || !tryNewDot?.classicRedirect?.dismissed) {
+        if (!tryNewDot?.classicRedirect?.dismissed) {
+            return false;
+        }
+
+        const dismissedReasons = tryNewDot.classicRedirect?.dismissedReasons;
+        let timestampToCheck: Date = tryNewDot.classicRedirect?.timestamp;
+
+        if (dismissedReasons && dismissedReasons.length > 0) {
+            const earliestReason = dismissedReasons.reduce((earliest, current) => {
```

Comment:
this should be latest not earliest, we want to know lastTime  user redirected from ND -> OD and compare if 30 days against that https://github.com/Expensify/App/pull/73436#issuecomment-3518179809